### PR TITLE
Extract ESIntegTestCase#prepareSearch

### PR DESCRIPTION
**Pull request by DaveCTurner**
Relates #101172

### CODE REVIEW

**Comment by DaveCTurner** [comment]
This was purely a mechanical search-and-replace job except:

- one of the overloads of `org.elasticsearch.xpack.ml.job.persistence.MockClientBuilder#prepareSearch` was unused, removed it
- renamed `org.elasticsearch.index.mapper.TokenCountFieldMapperIntegrationIT#prepareSearch` to `prepareTokenCountFieldMapperSearch` to avoid the name collision

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java [hunk 1]
```diff
@@ -1095,6 +1095,10 @@ public void waitForDocs(final long numDocs, final BackgroundIndexer indexer) thr
         }, maxWaitTimeMs, TimeUnit.MILLISECONDS);
     }
 
+    public static SearchRequestBuilder prepareSearch(String... indices) {
+        return prepareSearch(indices);
```
Comment (original-brownbear) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java:1099]: `client().prepareSearch()` I guess? :)

**Review by original-brownbear — APPROVED** [review]
LGTM after you fix the stack overflow (I think?) :)

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java (hunk 1 above)
Comment (DaveCTurner) [reply test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: D'oh

**Pull request merged**